src/library/scala/collection/generic/IsImmutableMapLike.scala

```diff
@@ -0,0 +1,45 @@
+package scala.collection
+package generic
+
+/**
+  * Type class witnessing that a collection type `Repr` has
+  * a conversion to `immutable.MapOps[K, V, _, _]`.
+  *
+  * @see [[scala.collection.generic.IsIterableLike]]
+  */
+trait IsImmutableMapLike[Repr] extends IsMapLike[Repr] {
+
+  // Convenient intermediate type definition to satisfy type bounds
+  protected type CC[X, +Y] <: Iterable[(X, Y)] with immutable.MapOps[X, Y, CC, _]
+  type C <: immutable.MapOps[K, V, CC, C]
+
+  /** A conversion from the type `Repr` to `immutable.MapOps[K, V, CC, C]` */
+  override def apply(c: Repr): immutable.MapOps[K, V, CC, C]
+
+}
+
+object IsImmutableMapLike {
+
+  // 1. Map collections
+  implicit def mapIsImmutableMapLike[CC0[X, +Y] <: Iterable[(X, Y)] with immutable.MapOps[X, Y, CC0, CC0[X, Y]], K0, V0]
+    : IsImmutableMapLike[CC0[K0, V0]] { type K = K0; type V = V0; type CC[X, +Y] = CC0[X, Y]; type C = CC0[K, V] } =
+    new IsImmutableMapLike[CC0[K0, V0]] {
+      type K = K0
+      type V = V0
+      type CC[X, +Y] = CC0[X, Y]
+      type C = CC0[K, V]
+      def apply(c: CC0[K0, V0]): immutable.MapOps[K0, V0, CC, C] = c
+    }
+
+  // 2. Sorted Map collections
+  implicit def sortedMapIsImmutableMapLike[CC0[X, +Y] <: immutable.Map[X, Y] with immutable.SortedMapOps[X, Y, CC0, CC0[X, Y]], K0, V0]
```

Comment:
Why is this necessary? `SortedMap` extends `Map`.

Reply:
Because `CC0[X, Y] <: SortedMap[X, Y]` gives us `CC0[X, Y] <: MapOps[X, Y, Map, CC0[X, Y]]`, whereas the `mapIsImmutableMapLike` definition requires `CC0[X, Y] <: MapOps[X, Y, CC0, CC0[X, Y]]` (note the third type parameter applied to `MapOps`).